### ISSUE TITLE
Looking for maintainers

### ISSUE THREAD

**Issue by guumaster** [mongoosastic/mongoosastic#457]
I think I'm one of the only maintainers active here. But I'm not using Mongoose nor Elastic lately, so it is hard for me to keep working on this.

We haven't published a new version for a long time. Our tests are not green since months. 

Our pending task are: 

- [x] Update/fix the test suite and make it pass on Travis CI.
- [x] Review/merge/close old PRs and issues.
- [x] Publish a new version

If you can help the project with all this, and we get to publish a new version, I'll add you as a permanent contributor/maintainer.

 **If there are any volunteer to fox tests, review issues and keep evolving this, reply here or DM me on twitter @guumaster.**

**Update**: Finally we got the tests fixed and a new version released. Huge thanks to @AxelJunker for all the efforts!.

**Comment by shubham-agrawal** [comment]
Hey I just saw this. I am still using mongoosastic. I have a decent idea of the codebase too as I had to sit and debug it once. So I won't mind maintaining it.

**Comment by guumaster** [comment]
Great news @shubham-agrawal! First thing needed is to fix the test and make the CI pass green. Any help would be much appreciated.

**Comment by shubham-agrawal** [comment]
Sure I will do it over the weekend.

**Comment by qballer** [comment]
I've just discovered this lib, being doing the heavy lifting myself. I might be interested in helping out. I also have ideas, like getting sql to  elastic solution. What do you think? Elasticity as a name.

**Comment by ankit307** [comment]
I would surely like to contribute. I am using this plugin in my 3 of project to sync up with elasticsearch and it need to be maintained now. Please count me In

**Comment by guumaster** [comment]
Hi @ankit307, all help is welcome. The plan is simple, fix the tests and then try to upgrade versions to keep up with latest `elasticsearch` and `mongoose` versions.

**Comment by AxelJunker** [comment]
@guumaster Hi! I am not too familiar on how to collaborate on OS-projects but I forked mongoosastic because I needed to fix some urgent issues I had in my own project. Now I have fixed almost all tests for elasticsearch@15.4.1 and mongoose@5.5.1 (those are the versions I use in my own project). 

1. How can I share my fixes?
2. How can I test my fixes against different mongoose- and elasticsearch versions? (without having to npm remove/install them)

**Comment by guumaster** [comment]
Hi @AxelJunker, great news about the tests. 

1- You can share your fixes by making a pull request from your fork to this one. There is a page detailing how to do [a pull request from a fork](https://help.github.com/en/articles/creating-a-pull-request-from-a-fork). Let me know if you need more help after reading that.

2- Not sure if you can do this _without_ remove/install deps. We can think about this for future releases though, maybe adding different branches and launching more TravisCI pipelines for different setups.

Thanks for reaching out, hope to see your fixes merged soon!

**Comment by wozniakm** [comment]
Hey! I'm also willing to help as I'm going to use mongoosastic in my project and I see potential improvements that I would like to add. Can you please create a pull request with your changes @AxelJunker?

**Comment by AxelJunker** [comment]
@wozniakm Great! There is a closed and merged [pull request #469](https://github.com/mongoosastic/mongoosastic/pull/469) for the develop branch. However, I've made some minor edits since then so I created a [new pull request #472](https://github.com/mongoosastic/mongoosastic/pull/472).

**Comment by guumaster** [comment]
Also, when the CI pass the tests I would make a new release.

**Comment by AxelJunker** [comment]
According to the CI logs, there seems to be some connection issues I don't get when testing locally.

**Comment by curioustushar** [comment]
@guumaster @AxelJunker Please let me know if I could help this product in some way. I've made a similar application as part of my day job, which syncs MongoDB data to Elasticsearch. 

@AxelJunker Yeah, you're right there some connection issues.

We get **Elasticsearch WARNING: No living connections**, I think we should have a check after running the docker to make sure that ES started and is reachable. If we add curl request to ES, we can get insights and also we should give some wait time for ES to be accessible before starting the tests.

What are your thoughts on that?

**Comment by AxelJunker** [comment]
@curioustushar Yeah, I added this line to the travis config ([source](https://code.dblock.org/2017/02/10/installing-starting-and-waiting-for-elasticsearch-in-travis-ci.html)):

`until curl --silent -XGET --fail http://localhost:9200; do printf '.'; sleep 1; done`

and it works now, thanks for the suggestion!

I also fixed the last failing test, so we should now be able to merge #472 to develop and master! @guumaster

**Comment by curioustushar** [comment]
@AxelJunker Glad that it worked. If there's anything else I can do to help, please let me know.

**Comment by AxelJunker** [comment]
@curioustushar Since the tests now work, I think the next step would be to try different versions of Elasticsearch, elasticsearch-js, and mongoose. I have only tested using Elasticsearch@6.4.2, elasticsearch-js@15.2.0, and mongoose@5.5.1.

Maybe @guumaster has something to say about this as well.

**Comment by guumaster** [comment]
Thanks to the great work from @AxelJunker we have green pipeline, now we can think of upgrading deps versions maintaining the passing tests.

**Comment by Henry-Asante** [comment]
I am about to use mongoosastic, I would love to join as a permanent maintainer as I believe it's a great project.

**Comment by guumaster** [comment]
A good start would be to fix the open PR #485 and make that tests pass to merge the new deps versions.

**Comment by guumaster** [comment]
@AxelJunker about different versions. I've setup a `renovate` bot to open PRs automatically with new deps udpates. As soon as the new versions have passing tests I plan to merge them and release a `mongoosastic` version. 

That's what you mean? Or keeping a table of compatibility? I'm not sure how to change tests to support multiple version at the same time.

**Comment by Henry-Asante** [comment]
> A good start would be to fix the open PR #485 and make that tests pass to merge the new deps versions.

Alright Noted!

**Comment by AxelJunker** [comment]
@guumaster I don't know exactly what I mean, hehe. But the renovate bot is really good news! I agree that it would be difficult to support multiple versions at the same time. So maybe we only support the latest stable release of the deps and let the renovate bot tell us when something is broken?

Another thing we can discuss is that the [elasticsearch.js client](https://github.com/elastic/elasticsearch-js-legacy) we use is now legacy. The [new client](https://github.com/elastic/elasticsearch-js) has many breaking changes. Should we upgrade to the new client?

**Comment by sherif-hafiz** [comment]
@guumaster are you still looking for contributors?

**Comment by guumaster** [comment]
Yes. The next goal would be to migrate to the new  elastic library and get it to support version 7. Any help is welcome

**Comment by sherif-hafiz** [comment]
I more than happy to contribute

**Comment by lakshmipriyamukundan** [comment]
I'm aware about mongoose and 'elasticsearch' module and working over for 3 years and hope I too can contribute

**Comment by LukeeeeBennett** [comment]
Happy to help

**Comment by Hisham-TK** [comment]
I would love to contribute,
I use mongoose for more than 1 year at a very advanced level,
and ElasticSearch not too much but my income projects related to ES.
I think i will do my best.

**Comment by khalilmansouri** [comment]
I am on working one project includes mongoosastic, I would love to contribute.

**Comment by jgcmarins** [comment]
cc @rturk

**Comment by sunny-shu** [comment]
@guumaster 
Are you still recruiting maintainers?
if yes,I am more free，i can take a try。

**Comment by guumaster** [comment]
@sunny-shu you are welcome to start maintaining this repo. First task would be to make tests pipelines pass, future task includes fix issues, maybe typescript re-writes. But tests are always main priority.

**Comment by sunny-shu** [comment]
> @sunny-shu you are welcome to start maintaining this repo. First task would be to make tests pipelines pass, future task includes fix issues, maybe typescript re-writes. But tests are always main priority.

ok, i will take a try.

**Comment by RaphaelHadjadj** [comment]
Thanks for this great package. I'm also interested by re-writes using TypeScript. I'm currently using your package, I don't have a lot of time, but I could try to rewrite it using TypeScript or make a definition file.

**Comment by ShettyAshwin** [comment]
Hey I am interested, please let me know how can help

**Comment by zsy2053** [comment]
Hi I'm interested, please let me know how to help

**Comment by subhamg** [comment]
Hey! I have a decent idea about the mongoose and elastic search. I have been using both for quite a while. Let me know how can I help you. Thanks

**Comment by subhamg** [comment]
Hi @guumaster, I want to convert the complete code into typescript. There are many developers using this library and are actually looking for types. Please let me help you with this. Thanks.

**Comment by rap2hpoutre** [comment]
This thread is not super active, so we created one more lib 🙄 (sorry 😅): https://github.com/selego/mongoose-elastic, maybe it could help some users, we use it in production.

**Comment by yahiakr** [comment]
Hi @guumaster @meabed 
I was working in the few past week on:

* immigrating to Typescript, 
* spliting the code so it would be more readable, 
* re-writing all the tests in Typescript, 
* defining the types for mongoose documents/models
* Imegrate to the new version of @elastic/elasticsearch-js & mongoose
* fix bugs related to bulking stream

Here's my repo: https://github.com/yahiakr/mongoosastic-plus

I really loved the package, and i would join you as a maintainer. If i can help with anything, please let me know.

**Comment by meabed** [comment]
> @meabed code seems great. One failing test is something that we can just skip and open an issue specific for that use case. Feel free to open a PR to develop branch and I'll try to merge it as soon as possible. Thanks for your hard work.
Amazing - I will prepare the PR shortly!


> Hi @guumaster @meabed I was working in the few past week on:
> 
> * immigrating to Typescript,
> * spliting the code so it would be more readable,
> * Immigrate to JEST as a testing framework
> * re-writing all the tests in Typescript,  (All the tests are exactly the same as the current version, and all are passed, i can check with your current tests to recognize the test failing cause).
> * defining the types for mongoose documents/models
> * Imegrate to the new version of @elastic/elasticsearch-js & mongoose
> * fix bugs related to bulking stream
> 
> Here's my repo: https://github.com/yahiakr/mongoosastic-plus
> 
> I really loved the package, and i would join you as a maintainer. If i can help with anything, please let me know.

@yahiakr Thats great!! would you have a chance to look at the way i have migrated it as well and see if there improvements there - so we maybe combine all the effort in one PR to merge it back here?

Thank you

**Comment by yahiakr** [comment]
Hi @meabed,
Of course! I will see if i can help with anything so we can make one major PR.
Thanks for your work and your efforts. I will let you know once i go through your repository.

**Comment by guumaster** [comment]
Thanks and Welcome the two new maintainers @yahiakr and @meabed :tada:

I'm closing this for now.
